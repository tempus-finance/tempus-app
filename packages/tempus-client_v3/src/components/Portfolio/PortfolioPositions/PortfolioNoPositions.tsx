import { FC, useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import { ActionButton, Typography } from '../../shared';

const PortfolioNoPositions: FC = () => {
  const navigate = useNavigate();
  const { t } = useTranslation();

  const handleBrowseMarketsClick = useCallback(() => {
    navigate('/');
  }, [navigate]);

  return (
    <div className="tc__portfolio-positions__empty">
      <Typography variant="subtitle" weight="bold">
        {t('PortfolioPositions.titleNoPositions')}
      </Typography>
      <Typography variant="subtitle">{t('PortfolioPositions.descriptionNoPositions')}</Typography>
      <ActionButton
        labels={{ default: t('PortfolioPositions.buttonBrowseMarkets') }}
        variant="primary"
        size="large"
        onClick={handleBrowseMarketsClick}
      />
    </div>
  );
};

export default PortfolioNoPositions;
