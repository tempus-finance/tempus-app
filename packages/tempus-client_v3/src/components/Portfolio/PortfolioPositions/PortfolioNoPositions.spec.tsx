import { fireEvent, render, RenderResult } from '@testing-library/react';
import { createMemoryHistory, History } from 'history';
import { Router } from 'react-router-dom';
import PortfolioNoPositions from './PortfolioNoPositions';

const subject = (): [RenderResult, History] => {
  const history = createMemoryHistory();
  history.push('/portfolio');

  const renderedSubject = render(
    <Router location={history.location} navigator={history}>
      <PortfolioNoPositions />
    </Router>,
  );

  return [renderedSubject, history];
};

describe('PortfolioNoPositions', () => {
  it('renders empty page when the user has no positions', () => {
    const [{ container }] = subject();

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('redirect when the user clicks on "Browse Markets" button', () => {
    const [{ getByRole }, history] = subject();
    const browseMarketsButton = getByRole('button');

    expect(browseMarketsButton).not.toBeNull();

    fireEvent.click(browseMarketsButton);

    expect(history.location.pathname).toBe('/');
  });
});
