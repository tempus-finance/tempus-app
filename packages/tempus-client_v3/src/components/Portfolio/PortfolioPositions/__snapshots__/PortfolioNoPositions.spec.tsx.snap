// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioNoPositions renders empty page when the user has no positions 1`] = `
<div>
  <div
    class="tc__portfolio-positions__empty"
  >
    <div
      style="font-style: normal; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      No Current Positions
    </div>
    <div
      style="font-style: normal; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Your portfolio is empty. Letâ€™s change that for you.
    </div>
    <button
      class="tc__btn tc__actionButton tc__actionButton-border-primary-large tc__actionButton-background-primary-large tc__actionButton-hover-primary-large tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-large "
      data-selected="false"
      type="button"
    >
      <div
        class="tc__actionButton-status-label"
      >
        <div
          style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(255, 255, 255); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
        >
          Browse Markets
        </div>
      </div>
    </button>
  </div>
</div>
`;
